pkg/kvstore/etcd.go

```diff
@@ -0,0 +1,372 @@
+//
```

Comment:
Right. We could probably have a bit more focused tests on each module, depending on the value they provide. In this case it'd be very useful to test high level correctness properties that ought to be guaranteed. Things like ensuring no duplicate IDs, ensuring no read-modify-write races, etc. Also important to write tests that exercise the code concurrently and run them with the builtin data race detector on as part of CI.